hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestingUtil.java

```diff
@@ -3642,6 +3662,110 @@ public boolean evaluate() throws IOException {
+        }
+        return getAdmin().tableExists(tableName) && totalRegionCount == prefetchedRegionCount.get();
+      }
+    };
+  }
+
+  /**
+   * Returns a {@Link Predicate} for checking that at least one region for the table is prefetched
+   */
+  public Waiter.Predicate<IOException> predicateAtLeastOneRegionIsPrefetchedOnServer(
+    final TableName tableName, final ServerName serverName) {
+    return new ExplainingPredicate<IOException>() {
+      @Override
+      public String explainFailure() throws IOException {
+        return "No Regions for table " + tableName.getNameAsString() + " prefetched on server "
+          + serverName.getAddress();
+      }
+
+      @Override
+      public boolean evaluate() throws IOException {
+        List<HRegion> regions = getMiniHBaseCluster().getRegions(tableName);
+        AtomicInteger prefetchedRegionCount = new AtomicInteger();
+        ServerMetrics sm =
+          getMiniHBaseCluster().getClusterMetrics().getLiveServerMetrics().get(serverName);
+        for (HRegion r : regions) {
+          sm.getRegionMetrics().forEach((rn, rm) -> {
+            if (
+              r.getRegionInfo().getRegionNameAsString().equals(rm.getNameAsString())
+                && rm.getCurrentRegionCachedRatio() == 1.0f
+            ) {
+              prefetchedRegionCount.getAndIncrement();
+            }
+          });
+        }
+        return getAdmin().tableExists(tableName) && prefetchedRegionCount.get() > 0;
+      }
+    };
+  }
+
+  /**
+   * Returns a {@Link Predicate} for checking that more than half of the regions for the table are
+   * prefetched
+   */
+  public Waiter.Predicate<IOException>
+    predicateMajorityRegionsArePrefetched(final TableName tableName) {
+    return new ExplainingPredicate<IOException>() {
+      @Override
+      public String explainFailure() throws IOException {
+        return "No Regions for table " + tableName.getNameAsString() + " prefetched";
+      }
+
+      @Override
+      public boolean evaluate() throws IOException {
+        List<HRegion> regions = getMiniHBaseCluster().getRegions(tableName);
+        int totalRegionCount = regions.size();
+        AtomicInteger prefetchedRegionCount = new AtomicInteger();
+        for (HRegion r : regions) {
+          getMiniHBaseCluster().getClusterMetrics().getLiveServerMetrics().forEach((sn, sm) -> {
+            sm.getRegionMetrics().forEach((rn, rm) -> {
+              String regionNameAsString = r.getRegionInfo().getRegionNameAsString();
+              String regionString = rm.getNameAsString();
+              if (regionNameAsString.equals(regionString)) {
+                if (rm.getCurrentRegionCachedRatio() == 1.0f) {
+                  prefetchedRegionCount.getAndIncrement();
+                }
+              }
+            });
+          });
+        }
+        return getAdmin().tableExists(tableName)
+          && (float) prefetchedRegionCount.get() / totalRegionCount > 0.5f;
+      }
+    };
+  }
+
   /**
```

Comment:
These utility methods are all specific for this new balance function, so it should not be here in HBaseTestingUtil. Please move it to your tests classes. Maybe make an abstract parent class which all your tests would extend?